src/main/java/redis/clients/jedis/JedisSlotBasedConnectionHandler.java

```diff
@@ -44,9 +43,12 @@ public Jedis getConnection() {
         if (result.equalsIgnoreCase("pong")) return jedis;
 
         pool.returnBrokenResource(jedis);
-      } catch (JedisConnectionException ex) {
+      } catch (JedisException ex) {
         if (jedis != null) {
-          pool.returnBrokenResource(jedis);
+          try {
```

Comment:
@marcosnils  @serge-groshev 

First of all, the code block Marcos suggested is having a bug that it returns instance to pool which is return value of the method.
(I was confused at the same point, so be attention all!)

This method MUST ensure that it returns one Jedis instance if available.
(regarding to antirez's ruby implementation)
In order to achieve, it intentionally consumes any exceptions. 
Actually it can even be changed to consume any RuntimeException in defensive programming's eyes.

IMHO why @serge-groshev meets the bug is that we were not defensive. We just forgot to check other than JedisConnectionException.